### PR TITLE
dropNew overflow strategy

### LINKED ISSUE #17167: Add buffer overflow strategy that drops the new element

### PR DESCRIPTION
**Pull request by 2m**
Fixes #17167.

### CODE REVIEW

File: akka-docs-dev/rst/java/stream-rate.rst [hunk 1]
```diff
@@ -107,6 +107,11 @@ we want to be nice to jobs that has been waiting for long, then this option can
 
 .. includecode:: ../../../akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/StreamBuffersRateDocTest.java#explicit-buffers-droptail
 
+Instead of dropping the youngest element from the tail of the buffer a new element can be dropped without
```
Comment (ktoso) [inline akka-docs-dev/rst/java/stream-rate.rst]: Maybe add incoming? `the new incoming element can be... `

File: akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala [hunk 2]
```diff
@@ -55,6 +57,11 @@ object OverflowStrategy {
   def dropBuffer: OverflowStrategy = DropBuffer
 
   /**
+   * If the buffer is full when a new element arrives, drops the new element.
+   */
+  def dropNew: OverflowStrategy = DropNew
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala]: Initiating naming bikeshed :wink: dropNew sounds weird to me, is `dropIncoming` better or weirder?

**Comment by ktoso** [comment]
Naming aside, impl LGTM :-)

File: akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala (hunk 2 above)
Comment (2m) [inline akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala]: image

Both sound alright for me. `dropNew` is a tad shorted though.

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala]: I prefer dropNew

**Comment by drewhk** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala (hunk 2 above)
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala]: Ok, I'm fine with dropNew after sleeping on it ;)

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/OverflowStrategy.scala]: then I merge ;)

**Pull request merged**